engine/src/main/java/engine/object/GameObject.java

```diff
@@ -67,4 +69,12 @@ public int getTypeId() {
     public int hashCode() {
         return Objects.hash(id);
     }
+
+    public boolean isSolid() {
+        return solid;
+    }
+
+    public void setSolid(boolean solid) {
```

Comment:
lets use contructor. Its better because developer may forget to make object solid which may lead to costly bug. Constructor restrict him to from it